src/main/kotlin/uk/gov/gdx/datashare/services/AcquirersService.kt

```diff
@@ -131,6 +150,48 @@ class AcquirersService(
     }
   }
 
+  fun deleteAcquirerSubscription(subscriptionId: UUID): AcquirerSubscription {
+    val now = dateTimeHandler.now()
+    adminActionAlertsService.noticeAction(
+      AdminAction(
+        "Delete acquirer subscription",
+        object {
+          val subscriptionId = subscriptionId
+          val whenDeleted = now
+        },
+      ),
+    )
+    val subscription = acquirerSubscriptionRepository.save(
+      acquirerSubscriptionRepository.findByIdOrNull(subscriptionId)?.copy(
+        whenDeleted = now,
+      ) ?: throw AcquirerSubscriptionNotFoundException("Subscription $subscriptionId not found"),
+    )
+
+    acquirerSubscriptionEnrichmentFieldRepository
+      .findAllByAcquirerSubscriptionId(subscriptionId)
+      .forEach { deleteAcquirerSubscriptionEnrichmentField(it.id) }
```

Comment:
If this is what we want to there's a `deleteAllByAcquirerSubscriptionId` method in the repository

Reply:
Unless @oskarwilliams you're particularly keen to get an email for every field?